services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/search/oss/SearchManagerImpl.java

```diff
@@ -51,34 +50,37 @@ public void init() throws IOException {
     @Override
     public void documentChangeMessages(List<ChangeMessage> messages) throws TemporarilyUnavailableException {
         try {
-            List<BulkOperation> operations = new ArrayList<>(messages.size());
-            messages.stream().forEach(changeMessage -> {
-                Document document = translator.generateSearchDocumentIfNecessary(changeMessage);
+            List<BulkOperation> operations = messages.stream().map(message -> {
+                Document document = translator.generateSearchDocumentIfNecessary(message);
                 if (document.getType() == DocumentTypeNames.add) {
-                    operations.add(BulkOperation.of(op -> op
+                    return BulkOperation.of(op -> op
                             .index(idx -> idx
                                     .index(SearchConstants.OPEN_SEARCH_INDEX_NAME)
                                     .id(document.getId())
-                                    .document(document.getFields()))));
-
+                                    .document(document.getFields())));
                 } else {
-                    operations.add(BulkOperation.of(op -> op
+                    return BulkOperation.of(op -> op
                             .delete(idx -> idx
                                     .index(SearchConstants.OPEN_SEARCH_INDEX_NAME)
-                                    .id(document.getId()))));
+                                    .id(document.getId())));
                 }
-            });
+            }).collect(Collectors.toList());
+
+            BulkResponse response = openSearchClient.bulk(req -> req.operations(operations));
 
-            BulkResponse response = openSearchClient.bulk((new BulkRequest.Builder().operations(operations).build()));
+            // if any message fails to process we will throw exception to reprocess it.
+            boolean hasError = response.items().stream().filter(item -> item.error() != null)
+                    .peek(item -> log.error("Document {} has error {} with reason {}.", item.id(), item.error().type(),
+                            item.error().reason())).findAny().isPresent();
```

Comment:
findAny will stop at the first error. You have to consume all of them if you want to log all the errors